includes/Core/Util/WP_Context_Switcher.php

```diff
@@ -0,0 +1,124 @@
+<?php
+/**
+ * Class Google\Site_Kit\Core\Util\WP_Context_Switcher
+ *
+ * @package   Google\Site_Kit\Core\Util
+ * @copyright 2020 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://sitekit.withgoogle.com
+ */
+
+namespace Google\Site_Kit\Core\Util;
+
+/**
+ * Class for temporarily switching WordPress context, e.g. from admin to frontend.
+ *
+ * @since n.e.x.t
+ * @access private
+ * @ignore
+ */
+class WP_Context_Switcher {
+
+	/**
+	 * Stack of original contexts switched from.
+	 *
+	 * @since n.e.x.t
+	 * @var array
+	 */
+	private static $context_stack = array();
+
+	/**
+	 * Switches to WordPress frontend context if necessary.
+	 *
+	 * Context is only switched if WordPress is not already in frontend context. Context should only ever be switched
+	 * temporarily. Call the returned closure as soon as possible after to restore the original context.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return callable Closure that restores context and returns true if context was restored or false otherwise.
+	 */
+	public static function with_frontend_context() {
+		global $current_screen;
+
+		if ( ! is_admin() ) {
+			return function() {
+				return false;
+			};
+		}
```

Comment:
As mentioned in the other method, I think it would be better to replace the early return with a conditional around the `switch` method – the restore function would simply restore the current screen (essentially a no-op), unless something else had changed the screen, in which case this would restore it as expected.

I don't think it matters though in this case since `front` requests don't have screens. It only really matters in the admin case to preserve the current screen.